### ISSUE TITLE
Minor fixes

### ISSUE THREAD

**Issue by JosePizarro3** [FAIRmat-NFDI/nomad-simulations#126]
While I was debugging using the Wannier90 parser, I got some errors to be fixed (see [nomad-parser-wannier90#3](https://github.com/FAIRmat-NFDI/nomad-parser-wannier90/pull/3)):

- [ ] Change order of normalization for 1st `KSpace` and then its subsections (`KMesh` and `KLinePath`).
- [x] Fix warning in `spectral_profile.py` adding condition for `if self.name == 'ElectronicDensityOfStates'` for total DOS.
- [ ] Overwrite FermiLevel and call for `ElectronicBandStructure.normalize()` once DFT+TB workflow is resolved.
- [x] No idea why, but in the frontend, the `SinglePoint.inputs` and `outputs` section are not shown.
- [x] Fix README
- [x] Fix equality operator
- [x] Move testing files to subfolders to mimic the src structure
- [x] Delete copyright at the beginning of each file

**Comment by JosePizarro3** [comment]
@ladinesa (**once you feel better**): can you tell me your thoughts on the first todo point regarding the order of calling normalize? I can explain more in detail (but you can also read the issue in the parser).

Also, the last point: I do not know why, but the `SinglePoint` links for inputs and outputs, the reference `section` is not showing on the front end:

![Screenshot from 2024-09-23 16-10-28](https://github.com/user-attachments/assets/b539523d-09cc-49bf-bb1b-e7e5c477e7a9)

It only happens for `Input Model System` and `Outputs`, but for `Input Model Method` is properly showing, see:

![Screenshot from 2024-09-23 16-11-18](https://github.com/user-attachments/assets/61cb430e-54ca-4d5c-9855-9abf76eb0c0a)

Any ideas what is going on wrong? I can share some branch with you to test.

**Comment by ladinesa** [comment]
> Any ideas what is going on wrong? I can share some branch with you to test.

Did you generate the gui artifacts? it could be that the definition is simply not loaded.

**Comment by JosePizarro3** [comment]
Yeah I did, but I can double check again (or maybe do before ctrl+f5...). ModelMethod seems to be working.

**Comment by ladinesa** [comment]
> @ladinesa (**once you feel better**): can you tell me your thoughts on the first todo point regarding the order of calling normalize? I can explain more in detail (but you can also read the issue in the parser).
> 

You could also try to normalize 'reciprocal_lattice_vectors' in the sub-section, and put an if in the main section if this is already normalized. You can isolate the normalize function for the reciprocal lattice vector so you can easily call it in the child section,  something like self.m_parent.normalize_reciprocal_lattice_vector(..). The current normalizer level applies only for the sub sections in a section but I am not sure how we can do this to order generally and if this makes sense.

**Comment by JFRudzinski** [comment]
@ladinesa Can you comment if the unchecked TODOs at the top are still relevant or might be still relevant?

**Comment by ladinesa** [comment]
point 1 I am not sure if this is already resolved somehow. I can look at points 3 and 4

**Comment by JFRudzinski** [comment]
Ok, thanks, you do not need to do it now, I will add it to the list of TODOs and then we can prioritize accordingly

**Comment by ladinesa** [comment]
I checked this again and it seems to work, at least for model_system.

<img width="1802" height="514" alt="Image" src="https://github.com/user-attachments/assets/e9b483f1-46fe-4aa8-8941-688292fed484" />

there is a currently a condition that there should not be any processing error otherwise the quantity is not shown. I tried to lift this condition in https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/2579

**Comment by ladinesa** [comment]
I understand now why it is showing for the current parsers, since mapping parser explicitly adds the m_def, otherwise gui cannot resolved it see  [issue](https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/issues/2164).

**Comment by ladinesa** [comment]
@JosePizarro3 or @ndaelman-hu can you enlighten me more about overwriting the Fermi level? where do i get the new value e.g in the case of dft-tb workflow.

**Comment by ladinesa** [comment]
@ndaelman-hu can you please have a look at issue 1.

**Comment by JosePizarro3** [comment]
> [@JosePizarro3](https://github.com/JosePizarro3) or [@ndaelman-hu](https://github.com/ndaelman-hu) can you enlighten me more about overwriting the Fermi level? where do i get the new value e.g in the case of dft-tb workflow.

There was some talk about it: https://github.com/nomad-coe/nomad-simulations/pull/201#discussion_r2230344013

Can you use `ChemicalPotential`? The Fermi level is nothing but the chem pot at 0K.

**Comment by JosePizarro3** [comment]
https://github.com/nomad-coe/nomad-simulations/blob/develop/src/nomad_simulations/schema_packages/properties/thermodynamics.py#L130

I am a bit confused that there is a `fermi_energy` under `ChemicalPotential` tho... maybe we should have handled this differently.

**Comment by JFRudzinski** [comment]
> https://github.com/nomad-coe/nomad-simulations/blob/develop/src/nomad_simulations/schema_packages/properties/thermodynamics.py#L130
> 
> I am a bit confused that there is a `fermi_energy` under `ChemicalPotential` tho... maybe we should have handled this differently.

I think most things under thermodynamics.py are quite preliminary. From what you said, perhaps:

FermiLevel(ChemicalPotential):
...
def normalize():
   # set T=0

?

**Comment by ndaelman-hu** [comment]
> While I was debugging using the Wannier90 parser, I got some errors to be fixed

Firstly, could we re-run these tests?

> Change order of normalization for 1st KSpace and then its subsections (KMesh and KLinePath).

`KSpace.reciprocal_lattice_vectors` define the subsections `KSpace.k_mesh` and `KSpace.k_line_path`.

> Overwrite FermiLevel and call for ElectronicBandStructure.normalize() once DFT+TB workflow is resolved.

@mohammadnakhaee Could you comment on this?

**Comment by ndaelman-hu** [comment]
> https://github.com/nomad-coe/nomad-simulations/blob/develop/src/nomad_simulations/schema_packages/properties/thermodynamics.py#L130
> 
> I am a bit confused that there is a `fermi_energy` under `ChemicalPotential` tho... maybe we should have handled this differently.

I'll take a look.

**Comment by mohammadnakhaee** [comment]
> Overwrite FermiLevel and call for ElectronicBandStructure.normalize() once DFT+TB workflow is resolved.
@mohammadnakhaee Could you comment on this?

I think Alvin has more detailed knowledge about the parsers and workflows here. @ladinesa, could you also comment?

**Comment by ndaelman-hu** [comment]
> > Overwrite FermiLevel and call for ElectronicBandStructure.normalize() once DFT+TB workflow is resolved.
> > [@mohammadnakhaee](https://github.com/mohammadnakhaee) Could you comment on this?
> 
> I think Alvin has more detailed knowledge about the parsers and workflows here. [@ladinesa](https://github.com/ladinesa), could you also comment?

Alvin actually came with this question to us. I hoped you might know given your involvement in some of the parsers.

**Comment by JosePizarro3** [comment]
> > > Overwrite FermiLevel and call for ElectronicBandStructure.normalize() once DFT+TB workflow is resolved.
> > > [@mohammadnakhaee](https://github.com/mohammadnakhaee) Could you comment on this?
> > 
> > 
> > I think Alvin has more detailed knowledge about the parsers and workflows here. [@ladinesa](https://github.com/ladinesa), could you also comment?
> 
> Alvin actually came with this question to us. I hoped you might know given your involvement in some of the parsers.

The problem comes from some TB approximations, in which the Fermi level is ill-defined, even thought some codes might report it (e.g., Wannier90 has a `fermi level` value in the output file, but this does not make any sense). What people tend to use is the Fermi level reported by the DFT side instead. 

This is what happens in Wannier90.
